### ISSUE TITLE
Base quality and consensus generation

### ISSUE THREAD

**Issue by nriddiford** [gear-genomics/tracy#58]
First off - thanks for all the great work on tracy. It's quite amazing to me how few tools there are for performing trace file assembly - so thanks for filling this void with a very nice tool! 

I have been using tracy quite a bit recently to assemble trace files, and perform variant calling relative to a reference sequence. Generally, this seems to work very well using tracy, but I have a question (related to [a previous issue](https://github.com/gear-genomics/tracy/issues/50#issue-1138781991) ) on the interplay between base-call confidence (on the chromatogram), and consensus formation. 

I'm seeing incorrect consensus calls being made for a particular base where one of the trace files contains a low-confidence call and the other a high confidence call. From what I understand (based on[ your previous explanation](https://github.com/gear-genomics/tracy/issues/50#issuecomment-1041526066)) tracy does not use the base quality from the chromatogram, and I guess just choses on base over the other when there's a disagreement?

Here's what I'm seeing: 

<img width="1078" alt="Screenshot 2022-04-06 at 16 19 32" src="https://user-images.githubusercontent.com/13079357/161996413-04b62e4c-22df-4940-91ca-b0514c0fbd1b.png">

This shows 2 trace files in Geneious. When I assemble these using `tracy assemble --format fastq --inccons trace1.ab1 trace2.ab1` the resulting consensus contains insertions at both positions highlighted in red. This is strange to me - the base quality in `trace 2` is clearly higher than in `trace 1`.

Is this expected behaviour? 

Thanks for any help!

**Comment by blex-max** [comment]
@tobiasrausch forgive me for pinging you, but are you intending to respond to this?

**Comment by tobiasrausch** [comment]
For `tracy assemble` it's a simple majority vote. If you have, for instance, 3 traces and 2 support a gap `-` and 1 a nucleotide then the gap `-` is chosen. Ties are arbitrarily broken and `tracy assemble` does not take into account qualities at the moment. For the pairwise case, `tracy consensus` does use the qualities but gaps don't have any to begin with.  Therefore, for `tracy consensus` it depends on whether you use `-i` or not.

**Comment by tobiasrausch** [comment]
Indeed, it's a 50:50 chance in theory but in order to make the algorithm deterministic the code currently favours nucleotides over gaps.

**Comment by blex-max** [comment]
Is this likely to change? As @nriddiford says, it seems a shame to have a 50% chance to incorporate a low quality base when the information is available to make the better call.
